lnwallet/size.go

```diff
@@ -0,0 +1,117 @@
+package lnwallet
+
+import (
+	"github.com/roasbeef/btcd/blockchain"
+)
+
+const (
+	WitnessFactor = blockchain.WitnessScaleFactor
+	MaxTransactionCost = blockchain.MaxBlockCost
+
+	// Below you can see the structure of a Commitment transaction which
+	// will be broadcasted in case of force closure of channel. We should
+	// estimate its size rather than cooperative closure transaction,
+	// because this transaction is bigger.
+
+	// The weight(cost), which is different from the !size! (see BIP-141),
+	// is calculated as:
+	// Weight = 4 * BaseSize + WitnessSize (weight).
+	// BaseSize - size of the transaction without witness data (bytes).
+	// WitnessSize - witness size (bytes).
+	// Weight - the metric for determining the cost of the transaction.
+
+	// PW2SH:
+	//	- OP_0: 1 byte
+	//	- WitnessScriptSHA256: 32 bytes
+	P2SHSize = 1 + 32
+
+	// P2PKH:
+	//	- OP_0: 1 byte
+	//	- PublicKeyHASH160: 20 bytes
+	P2PKHSize = 1 + 20
+
+	// MultiSig:
+	//	- OP_2: 1 byte
+	//	- pubKeyAliceLength: 1 byte
+	//	- pubKeyAlice: 33 bytes
+	//	- pubKeyBobLength: 1 byte
+	//	- pubKeyBob: 33 bytes
+	//	- OP_2: 1 byte
+	//	- OP_CHECKMULTISIG: 1 byte
+	MultiSigSize = 1 + 1 + 33 + 1 + 33 + 1 + 1
+
+
+	// Witness:
+	//	- nil byte: 1 byte
+	//	- sigAlice: 73 bytes
+	//	- sigBob: 73 bytes
+	//	- WitnessScript (MultiSig)
+	WitnessSize = 1 + 73 + 73 + MultiSigSize
```

Comment:
This also misses the two additional var-ints on the wire which specify that there's only 1 item on the witness stack, and also the size of the witness element itself.